File: LuhnAlgorithm.py


def cardType(n):
    s="Visa"
    if n[0]=="5":
        s="MasterCard"
    return s;
def checkSum(card):
    chksum=0
    f=0
    for x in card:
        if f==0:
            temp= (ord(x)-48)*2
            if temp>9:
                a = str(temp)[0]
                b = str (temp)[1]
                temp= (ord(a)-48) + (ord(b)-48)
            chksum+=temp
            f=1
        else:
             chksum+= (ord(x)-48)
             f=0
       
    
    return chksum;

print("#", "  Credit Card #", "     Type", "      Checksum", " Valid?",'\n')
i=1


file= open('credit.txt','r')
lines= file.readlines()



for index in lines:
        
        ty= cardType(index.strip())
        c= checkSum(index.strip())
        valid= "NO"
        if(c%10 == 0):
            valid="YES"
        if(ty=="Visa"):
                print(i,index.strip(),"   ",ty,"      ",c,"      ",valid,'\n')  
        else: 
             print(i,index.strip(),"   ",ty,c,"      ",valid,'\n')
        i=i+1
        


   

File: LuhnAlgorithm.cpp
#include <bits/stdc++.h>
using namespace std;
#define ll long long int
//#define khawla ios::sync_with_stdio(false);cin.tie(NULL);cout.tie(NULL);
int checkSum(string card){
  int checksum=0;
  for (int i = 0; i < card.size(); i+=2) {
    int temp = (card[i]-'0')*2;
    if(temp>9){
      int x= temp/10;
      int y= temp%10;
      card[i]= (x+y)+'0';
    }else{
      card[i]= temp+'0';
    }
  }
  for (int i = 0; i < card.size(); i++) {
    checksum+= card[i]-'0';
  }
  return checksum;
}
int main()
{

  //khawla


  string card;

  ifstream MyReader("credit.txt");

  int i =1;
  while (getline (MyReader, card)) {
   if (i==1) {
     cout<<"#"<<"  "<<"Credit Card #"<<"      "<<"Type "<<"     "<<"Checksum"<<"  "<<"Valid?"<<'\n';
     cout<<"--------------------------------------------------"<<'\n';
   }
  string valid="NO";
  string type="Visa";
  int sum= checkSum(card);
  if (sum%10 == 0) {
    valid= "YES";
  }
  if (card[0]=='5') {
    type="MasterCard";
    cout<<i<<"  "<<card<<"   "<<type<<"  "<<sum<<"        "<<valid<<'\n';
  }else{
    cout<<i<<"  "<<card<<"   "<<type<<"        "<<sum<<"        "<<valid<<'\n';
  }



  i++;

  }


  MyReader.close();

    return 0;
}
